File: app/src/main/java/com/foreveross/atwork/component/scrollView/MaxHeightSwitchOrgScrollView.java
package com.foreveross.atwork.component.scrollView;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.ScrollView;

import com.foreveross.atwork.infrastructure.utils.DensityUtil;

/**
 * Created by dasunsy on 16/5/27.
 * todo 后期改成可配置
 */
public class MaxHeightSwitchOrgScrollView extends ScrollView {


    public MaxHeightSwitchOrgScrollView(Context context) {
        super(context);

    }

    public MaxHeightSwitchOrgScrollView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    @Override
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        Context context = getContext();
        if (null != context) {
            heightMeasureSpec = MeasureSpec.makeMeasureSpec(DensityUtil.dip2px(290), MeasureSpec.AT_MOST);
        }
        super.onMeasure(widthMeasureSpec, heightMeasureSpec);
    }
}

File: dependencies/w6s-ui-component/src/main/java/com/foreverht/workplus/ui/component/scrollView/MaxHeightSwitchOrgScrollView.kt
package com.foreverht.workplus.ui.component.scrollView

import android.content.Context
import android.util.AttributeSet
import android.view.View
import android.widget.ScrollView
import com.foreverht.workplus.ui.component.Util.Util

/**
 * Created by wuzejie on 2019/10/29.
 */

class MaxHeightSwitchOrgScrollView : ScrollView {


    constructor(context: Context) : super(context) {

    }

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {}

    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        var heightMeasureSpec = heightMeasureSpec
        val context = context
        if (null != context) {
            heightMeasureSpec = View.MeasureSpec.makeMeasureSpec(Util.dip2px(context, 400f), View.MeasureSpec.AT_MOST)
        }
        super.onMeasure(widthMeasureSpec, heightMeasureSpec)
    }
}